File: solutions/c/difference-of-squares/1/difference_of_squares.c
#include "difference_of_squares.h"
#include <stdint.h>
#include <math.h>
unsigned int sum_of_squares(unsigned int number){
    uint32_t suma=0;
    for(unsigned int i=1; i<=number; i++){
        suma+=pow(i, 2);
    }
    return suma;
}
unsigned int square_of_sum(unsigned int number){
    uint32_t suma=0;
    for(unsigned int i=0;i<=number;i++){
        suma+=i;
    }
    return pow(suma, 2);
}
unsigned int difference_of_squares(unsigned int number){
    unsigned int Suma_De_Cuadrados = sum_of_squares(number);
    unsigned int Cuadrado_De_La_Suma = square_of_sum(number);
    return Cuadrado_De_La_Suma - Suma_De_Cuadrados;
}

File: solutions/zig/difference-of-squares/1/difference_of_squares.zig
pub fn squareOfSum(number: usize) usize {
    var result: usize = 0;
    for (0..number+1) |i| {
        result += i;
    }
    result *= result; 
    return result;
}

pub fn sumOfSquares(number: usize) usize {
    const result = (number*(number+1)*((2*number)+1))/6;
    return result;
}

pub fn differenceOfSquares(number: usize) usize {
    const result = squareOfSum(number) - sumOfSquares(number);
    return result;
}
